stats/stats/src/charts/lines/txns_growth.rs

```diff
@@ -0,0 +1,130 @@
+use super::NewTxns;
+use crate::{
+    cache::Cache,
+    charts::{
+        find_chart,
+        insert::{insert_data_many, DateValue},
+        updater::{get_last_row, get_min_block_blockscout},
+    },
+    UpdateError,
+};
+use async_trait::async_trait;
+use entity::{chart_data, sea_orm_active_enums::ChartType};
+use sea_orm::{prelude::*, QueryOrder, QuerySelect};
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub struct TxnsGrowth {
+    cache: Mutex<Cache<Vec<DateValue>>>,
+}
+
+impl TxnsGrowth {
+    pub fn new(cache: Cache<Vec<DateValue>>) -> Self {
+        Self {
+            cache: Mutex::new(cache),
+        }
+    }
+}
+
+#[async_trait]
+impl crate::Chart for TxnsGrowth {
+    fn name(&self) -> &str {
+        "txnsGrowth"
+    }
+
+    fn chart_type(&self) -> ChartType {
+        ChartType::Line
+    }
+
+    async fn update(
+        &self,
+        db: &DatabaseConnection,
+        blockscout: &DatabaseConnection,
+        full: bool,
+    ) -> Result<(), UpdateError> {
+        let chart_id = find_chart(db, self.name())
+            .await
+            .map_err(UpdateError::StatsDB)?
+            .ok_or_else(|| UpdateError::NotFound(self.name().into()))?;
+        let min_blockscout_block = get_min_block_blockscout(blockscout)
+            .await
+            .map_err(UpdateError::BlockscoutDB)?;
+        let last_row = get_last_row(self, chart_id, min_blockscout_block, db, full).await?;
+        let mut data = {
+            let mut cache = self.cache.lock().await;
+            cache
+                .get_or_update(async move { NewTxns::read_values(blockscout, last_row).await })
+                .await?
+        };
+        data.sort();
+        let min_data = data.first().map(|v| v.date);
+        if let Some(to) = min_data {
+            let last_data: Option<DateValue> = chart_data::Entity::find()
+                .column(chart_data::Column::Date)
+                .column(chart_data::Column::Value)
+                .filter(chart_data::Column::ChartId.eq(chart_id))
+                .filter(chart_data::Column::Date.lt(to))
+                .order_by_desc(chart_data::Column::Date)
+                .into_model()
+                .one(db)
+                .await
+                .map_err(UpdateError::StatsDB)?;
+            let mut starting_sum = match last_data {
+                Some(last_data) => last_data
+                    .value
+                    .parse::<i64>()
+                    .map_err(|e| UpdateError::Internal(e.to_string()))?,
+                None => {
+                    if last_row.is_some() {
+                        tracing::warn!(
+                            chart_name = self.name(),
+                            last_row = ?last_row,
+                            "last_row is Some, but chart doesn't have any data until this row"
```

Comment:
1. this isn't something to warn about, because this is a regular thing, i.e. on clear db it's absolutely normal to have no data before
2. but I can see how this can be useful if something goes wrong, but I really think we should log "what we will do because condition", and not just "some condition": for example, log here "calculating growth from 0, because no old data was found". Ask yourself, can you interpret the log without looking at the code? Your message doesn't give any context: ok, chart doesn't have any data, so what? When in my message it's clear what we will do and why